### PR TITLE
[Security][Acl] enforce string identifiers

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14775 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Security/Acl/Domain/ObjectIdentity.php [hunk 1]
```diff
@@ -66,7 +66,7 @@ public static function fromDomainObject($domainObject)
             if ($domainObject instanceof DomainObjectInterface) {
                 return new self($domainObject->getObjectIdentifier(), ClassUtils::getRealClass($domainObject));
```
Comment (jakzal) [inline src/Symfony/Component/Security/Acl/Domain/ObjectIdentity.php]: We should probably cast an identifier to string here too.

Comment (xabbuh) [inline src/Symfony/Component/Security/Acl/Domain/ObjectIdentity.php]: I don't think so. In this case, returning a string is mandatory if you want to properly implement the `DomainObjectInterface`. We don't do type checks for scalars anywhere else. Do you think we should introduce that here?

Comment (jakzal) [inline src/Symfony/Component/Security/Acl/Domain/ObjectIdentity.php]: Nope. You're right.

**Comment by jakzal** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**